### ISSUE TITLE
[SC2086] Passing list known only at runtime to function as positional args

### ISSUE THREAD

**Issue by seanbright** [koalaman/shellcheck#2716]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or "online"): online
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

packages="one two three four"

check_packages() {
    for pkg; do
        echo "checking $pkg..."
    done
}

check_packages $packages

```

#### Here's what shellcheck currently says:

```
Line 11:
check_packages $packages
               ^-- SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean:
check_packages "$packages"
```

#### Here's what I wanted or expected to see:

The help on the wiki for POSIX sh suggests creating a function which works fine if you know the list of function arguments ahead of time (e.g. `check_packages one two three four`). In my case I don't know ahead of time what `$packages` will be. The wiki then goes on to say:

> To split on spaces but not perform glob expansion, POSIX has a `set -f` to disable globbing. You can disable word splitting by setting `IFS=''`.

I _think_ this (split on spaces without globbing) is what I want for my case but it's not clear to me how to put this suggestion into practice.

In short, I am trying to accomplish the above without disabling the shellcheck warning and without using arrays.
